tests/test_e2e.py

```diff
@@ -93,16 +96,19 @@ def setup_and_teardown(self):
             "grpc",
             pytest.param(
                 "http",
-                marks=pytest.mark.xfail(
+                # NOTE: skip because xfail was causing server to not get cleaned up by test in background
+                marks=pytest.mark.skip(
```

Comment:
Just for my own understanding, what do you mean by server not get cleaned up? Is there a dangling server process in the background? I didn't recall to see this when adding the test, so just wanted to learn more.

Reply:
Answered offline with example job that hung, but yes it appeared the server was dangling in the background even after unit test is logically complete, the server doesn't get cleaned up, and the test just hangs there until job timeout.